From 2e1ec9d85707cb29cbe055c7af75b6ea4335de5c Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Wed, 25 Sep 2024 11:30:02 +0800
Subject: [PATCH 3/3] services4: srvkm: mmap: use vma->vm_flags wrapper
 functions

kernel commit bc292ab00f6c7a(mm: introduce vma->vm_flags wrapper functions)
introduced wrapper functions to replace vm_flags setting. Here use vm_flags_set()
to compatible kernel v6.6.

(LOCAL REV: NOT UPSTREAM) - Wrlinux specific

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 services4/srvkm/env/linux/mmap.c | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/services4/srvkm/env/linux/mmap.c b/services4/srvkm/env/linux/mmap.c
index 69f23d0..f750332 100644
--- a/services4/srvkm/env/linux/mmap.c
+++ b/services4/srvkm/env/linux/mmap.c
@@ -744,7 +744,7 @@ DoMapToUser(LinuxMemArea *psLinuxMemArea,
 #if defined(PVR_MAKE_ALL_PFNS_SPECIAL)
 		if (bMixedMap)
 		{
-		        ps_vma->vm_flags |= VM_MIXEDMAP;
+			vm_flags_set(ps_vma, VM_MIXEDMAP);
 		}
 #endif
 	/* Second pass, get the page structures and insert the pages */
@@ -1056,18 +1056,7 @@ PVRMMap(struct file* pFile, struct vm_area_struct* ps_vma)
     PVR_DPF((PVR_DBG_MESSAGE, "%s: Mapped psLinuxMemArea 0x%p\n",
          __FUNCTION__, psOffsetStruct->psLinuxMemArea));
 
-    ps_vma->vm_flags |= VM_DONTDUMP;
-    ps_vma->vm_flags |= VM_IO;
-
-    /*
-     * Disable mremap because our nopage handler assumes all
-     * page requests have already been validated.
-     */
-    ps_vma->vm_flags |= VM_DONTEXPAND;
-    
-    /* Don't allow mapping to be inherited across a process fork */
-    ps_vma->vm_flags |= VM_DONTCOPY;
-
+    vm_flags_set(ps_vma, VM_DONTDUMP | VM_IO | VM_DONTEXPAND | VM_DONTCOPY);
     ps_vma->vm_private_data = (void *)psOffsetStruct;
     
     switch(psOffsetStruct->psLinuxMemArea->ui32AreaFlags & PVRSRV_HAP_CACHETYPE_MASK)
-- 
2.34.1

